test/integration/puppetlabs/services/master/environment_modules_int_test.clj

```diff
@@ -0,0 +1,392 @@
+           ;;    the 'test' environment was not flushed.
+           (purge-env-dir)
+           (purge-all-env-caches)
+           ;; add version that has multiple module versions
+           (let [prod-file (testutils/write-pp-file "class oneprod {}"
+                                                    "somemod"
+                                                    "prod"
+                                                    "production"
+                                                    "1.0.0")
+                 test-file (testutils/write-pp-file "class onetest {}"
+                                                    "somemod"
+                                                    "test"
+                                                    "test"
+                                                    "2.0.0")
+                 production-response-initial (get-env-modules "production")
+                 test-response-initial (get-env-modules "test")]
+             (is (= 200 (:status production-response-initial))
+                 (str
+                  "unexpected status code for initial production response"
+                  "response: "
+                  (ks/pprint-to-string production-response-initial)))
+             (is (= {"modules" [{"name" "somemod", "version" "1.0.0"}]
+                     "name" "production"}
+                    (response->module-info-map production-response-initial))
+                 "unexpected body for production response")
+             (is (= 200 (:status test-response-initial))
+                 (str
+                  "unexpected status code for initial test response"
+                  "response: "
+                  (ks/pprint-to-string test-response-initial)))
+             (is (= {"modules"[{"name" "somemod", "version" "2.0.0"}]
+                     "name" "test"}
+                    (response->module-info-map test-response-initial))
+                 "unexpected body for test response")
+
+             (testutils/write-pp-file "class oneflushprod {}"
+                                      "somemod"
+                                      "prod"
+                                      "production"
+                                      "1.0.0")
+             (testutils/write-pp-file "class oneflushtest {}"
+                                      "somemod"
+                                      "test"
+                                      "test"
+                                      "2.0.0")
+
+             (let [production-response-before-flush (get-env-modules
+                                                     "production")
+                   test-response-before-flush (get-env-modules
+                                               "test")]
+               (is (= 304 (:status production-response-before-flush))
+                   (str
+                    "unexpected status code for prod response after code change "
+                    "but before flush"))
```

Comment:
It seems like the following tests are failing. I kind of expected caching to work the same way as environment_classes with no changes on my end, so perhaps something it set up wrong here? For example:

```
FAIL in (environment-modules-integration-cache-enabled-test) (environment_modules_int_test.clj:258)
environment class cache invalidation for one environment
unexpected status code for prod response after code change but before flush
expected: (= 304 (:status production-response-before-flush))
  actual: (not (= 304 200))

lein test :only puppetlabs.services.master.environment-modules-int-test/environment-modules-integration-cache-enabled-test

FAIL in (environment-modules-integration-cache-enabled-test) (environment_modules_int_test.clj:262)
environment class cache invalidation for one environment
unexpected body for production response
expected: (empty? (:body production-response-before-flush))
  actual: (not (empty? "{\"modules\":[{\"name\":\"somemod\",\"version\":\"1.0.0\"}],\"name\":\"production\"}"))
```

I thought maybe the `purge-env-dir` needed to be updated but it looks like it completely deletes the environment folder which should include modules?